File: 04. Quick sort/main.py
def quick_sort(array):
    if len(array) < 2:
        return array
    pivot = array[0]
    less = [i for i in array[1:] if i < pivot]
    greater = [i for i in array[1:] if i > pivot]
    return quick_sort(less) + [pivot] + quick_sort(greater)


array = [2, 12, 201, 73, 1, 82, 47, 192, 17, 99, 104, 65, 50]
print(quick_sort(array))

File: 04. Quick sort/Main.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Main {
    public int[] sortArray(int[] nums) {
        if (nums.length < 2)
            return nums;
        int pivot = nums[0];

        int[] less = sortArray(Arrays.stream(nums).skip(1).filter(x -> x <= pivot).toArray());
        int[] high = sortArray(Arrays.stream(nums).skip(1).filter(x -> x > pivot).toArray());

        List<Integer> resultList = new ArrayList<>();

        for (int num : less) {
            resultList.add(num);
        }
        resultList.add(pivot);
        for (int num : high) {
            resultList.add(num);
        }
        int[] result = new int[resultList.size()];
        for (int i = 0; i < result.length; i++) {
            result[i] = resultList.get(i);
        }
        return result;
    }
}